src/createStreamingManager.ts

```diff
@@ -37,57 +37,58 @@ function mapConnectionState(state: RTCIceConnectionState): ConnectionState {
     }
 }
 
-function pollStats(peerConnection, onVideoStateChange, analytics) {
-    let videoStats = [] as SlimRTCStatsReport[];
-    let videoStatsStartIndex = 0;
-    let videoStatsLastIndex = 0;
-    let isPlayingFalseNumIntervals = 0;
-    let isPlaying: boolean;
-
-    return setInterval(() => {
-        const stats = peerConnection.getStats();
-        stats.then(result => {
-            result.forEach(report => {
-                if (report.type === 'inbound-rtp' && report.kind === 'video') {
-                    videoStatsLastIndex = videoStats.length - 1;
-                    if (report && videoStats[videoStatsLastIndex]) {
-                        const currBytesReceived = report.bytesReceived;
-                        const lastBytesReceived = videoStats[videoStatsLastIndex].bytesReceived;
-                        const prevPlaying = isPlaying;
-                        let videoStatsReport;
-
-                        isPlaying = currBytesReceived - lastBytesReceived > 0;
-                        isPlayingFalseNumIntervals = isPlaying ? 0 : ++isPlayingFalseNumIntervals;
-
-                        if (prevPlaying !== isPlaying) {
-                            if (isPlaying) {
-                                videoStatsStartIndex = videoStats.length;
-                                onVideoStateChange?.(StreamingState.Start, videoStatsReport);
-                            } else {
-                                const stats = videoStats.slice(videoStatsStartIndex);
-                                const previousStats =
-                                    videoStatsStartIndex === 0 ? undefined : videoStats[videoStatsStartIndex - 1];
-                                videoStatsReport = createVideoStatsReport(stats, previousStats);
-                                videoStatsReport = videoStatsReport
-                                    .sort((a, b) => b.packetsLost - a.packetsLost)
-                                    .slice(0, 5);
-                            }
-                        }
-
-
-                        if (!isPlaying && isPlayingFalseNumIntervals === 1) {
-                            analytics.track('agent-video', { event: 'stats', rtc: videoStatsReport });
-                        }
-
-                        if (!isPlaying && isPlayingFalseNumIntervals === 2) {
-                            onVideoStateChange?.(StreamingState.Stop, videoStatsReport);
-                        }
-                    }
-                    videoStats.push(report);
+function createVideoStatsAnalyzer() {
+    let lastBytesReceived = 0;
+
+    return (stats: RTCStatsReport) => {
+        stats.forEach(report => {
+            if (report.type === 'inbound-rtp' && report.kind === 'video') {
+                if (report) {
```

Comment:
if (report.type === 'inbound-rtp' && report.kind === 'video') {
                if (report) {

the order doesn't make sense

Reply:
I kept the original order, you're right. I'll add it to the condition